### PR TITLE
private-events

### PR DESCRIPTION
**Pull request by ashraffares**
Add three model :
- user model with devise and authentication and before action
- event model with validation of all event inputs presence true
- invitation  model with validation of all invitation inputs presence true

Create an association between the models
-user has_many created_events AND user has_many invitation AND user has_many attended_event through invitation.
-event belogs_to creator AND event has_many invitation AND has_many attendees through invitation.
- invitation belongs_to user AND invitation belongs_to event.

Add devise views and style them with bootstrap cdn.
Add all views necessary for show and update and index events and style them with bootstrap CDN.

Add tests with RSpec to the models
user - event - invitation

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  # GET /events/new
+  def new
+    @event = current_user.created_events.build
+  end
+
+  # GET /events/1/edit
+  def edit; end
+
+  # POST /events or /events.json
+  def create
```
Comment (GzimAsani) [inline app/controllers/events_controller.rb:26]: uses #build method on association, i.e. `current_user.events.build(event_params)` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/build_method_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index]
+  before_action :correct_user, only: %i[edit update destroy]
+
+  # GET /events or /events.json
+  def home; end
+
+  def index
```
Comment (GzimAsani) [inline app/controllers/events_controller.rb:8]: * [ ] lists past events, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [ ] lists ucomping events, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Status: Changes Required :red_circle: :recycle: 

### Hello @ashraffares, great job so far :clap: 

### To highlight:
- Very nice readme file
- Very professional commits


### To improve:

- [x] The event's show page displays the list of attendants ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/events_attendees_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)) -> _Currently your events show page does not show the list of attendants_ 
- [x] The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)) -> _Currently you do not have a user show page, kindly add it_
- [x] Events Controller / Create action needs update check the comment below :arrow_down: 
- [x] Events Controller / Index action needs update check the comment below :arrow_down: 
- [x] You should create a users controller and the show action should do the following:
1. list events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
2. list events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
3. list upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [x] Please implement the user as an Event attendee ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/attendee_assoc_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

_Please implement all the above mentioned requirements and ask for another code review whenver you are ready_

### **If you have any question related to the requirements please add a comment here and ping me on slack @GzimAsani**

# HAPPY CODING :computer: :desktop_computer:

**Comment by ashraffares** [comment]
Add invitation list
Add creator controller and info page to show past and upcoming and created event for the user
Add user link by his name he can access all his events in the navbar

**Review by NtwaliHeritier — APPROVED** [review]
### **STATUS: APPROVED✔✔**

Great work making the required changes.
You can merge.👏👍

Happy coding💻💻

**Pull request merged**